pkg/log/checkpoint/files_test.go

```diff
@@ -31,6 +31,8 @@ func TestFilesFound(t *testing.T) {
 
 	for _, tt := range tests {
 		t.Run("testFilesFound", func(t *testing.T) {
+			t.Parallel()
```

Comment:
Do we need to re-init tt here? We probably should, so we're consistent

Reply:
This is a good one, the lack of re-init made the test give a false pass.

The all above to `defer os.RemoveAll(tempDir)` is being executed before/during the parallel subtests and was deleting them temp directory ... which makes sense. 

Discovered a nifty little `t.Cleanup(func ())` built into the testing package. 

"Cleanup registers a function to be called when the test and all its subtests complete. Cleanup functions will be called in last added, first called order."

```
	t.Cleanup(func() {
		os.RemoveAll(tempDir)
	})
```